collector/lib/ConnTracker.cpp

```diff
@@ -76,14 +75,15 @@ IPNet ConnectionTracker::NormalizeAddressNoLock(const Address& address) const {
     return {};
   }
 
-  bool private_addr = !address.IsPublic();
+  // We want to keep private addresses and explicitely requested ones.
+  bool keep_addr = !address.IsPublic() || !detailed_networks_.Find(address).IsNull();
```

Comment:
Maybe we should keep this as separate checks? So it's a bit more explicit and gives a more flexibility in future checks, WDYT?
```suggestion
  bool private_addr = !address.IsPublic();
  bool detailed_addr = !detailed_networks_.Find(address).IsNull();
```

Reply:
I agree that having a very clear definition of when to keep addresses unaggregated is important. However, I think that `keep_addr` brings more meaning than expanding to `private_addr || do_not_aggregate_addr` everywhere it is used.

Reply:
That's good, thanks!